### PR TITLE
console proxy client sleep more generic

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #8692

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@weizhouapache @nvazquez , this is purely based on the description in #8692. Any advice on verification?

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java [hunk 1]
```diff
@@ -139,6 +134,11 @@ public void run() {
                                 connectionAlive = false;
                             }
                         }
+                        try {
+                            Thread.sleep(1);
+                        } catch (InterruptedException e) {
+                            s_logger.error("Error on sleep for vnc over websocket", e);
```
Comment (weizhouapache) [inline services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java:140]:
```suggestion
                            s_logger.error("Error on sleep for vnc sessions", e);
```

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Comment by kohrar** [comment]
A couple questions, if you don't mind:

1. Would it make sense from a latency-perspective to only wait 1ms if there's actually no data coming in? If there's a big chunk of data coming in, you might not want to wait 1ms to read ~200 bytes (which appears to be the maximum the NIO readNextBytes method would return)
2. Why 1ms? Could we raise this to reduce wasted CPU cycles even more? My testing showed the CPU went from 100% to 25%. This might still seem a bit excessive.

**Comment by DaanHoogland** [comment]
> A couple questions, if you don't mind:
> 
>     1. Would it make sense from a latency-perspective to only wait 1ms if there's actually no data coming in? If there's a big chunk of data coming in, you might not want to wait 1ms to read ~200 bytes (which appears to be the maximum the NIO readNextBytes method would return)
> 
>     2. Why 1ms? Could we raise this to reduce wasted CPU cycles even more? My testing showed the CPU went from 100% to 25%. This might still seem a bit excessive.

valid questions @kohrar , sounds like you are arguing for configurability. This is the simplest solution to the issue in your initial ticket. What you are asking now is a bigger (functional) change however. I would prefer to discuss and implement that separately.

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**